Source/bmalloc/bmalloc/TZoneHeapManager.h

```diff
@@ -147,18 +150,33 @@ class TZoneHeapManager {
 
     BINLINE static TZoneHeapManager& singleton()
     {
-        if (!theTZoneHeapManager)
+        if (!theTZoneHeapManager) {
             ensureSingleton();
+            theTZoneHeapManager->init();
+        }
         BASSERT(theTZoneHeapManager);
         return *theTZoneHeapManager;
     }
 
+    BINLINE static void initWithBucketSettings(unsigned smallSizeCount, unsigned largeSizeCount = 0, unsigned smallSizeLimit = 0)
+    {
+        RELEASE_BASSERT(!theTZoneHeapManager);
+        if (!theTZoneHeapManager) {
+            ensureSingleton();
+            RELEASE_BASSERT(theTZoneHeapManager);
+            theTZoneHeapManager->setBucketParams(smallSizeCount, largeSizeCount, smallSizeLimit);
+            theTZoneHeapManager->init();
+        }
+    }
+
     BEXPORT pas_heap_ref* heapRefForTZoneType(bmalloc_type* classType);
     BEXPORT void dumpRegisteredTypes();
 
 private:
     BEXPORT static void ensureSingleton();
 
+    BEXPORT void setBucketParams(unsigned smallSizeCount, unsigned largeSizeCount = 0, unsigned smallSizeLimit = 0);
```

Comment:
Do we really need to define the default values for the arguments since the default cases never happens?

Reply:
This is here if the caller only wants to set the small and large counts and not the max small size.  This will probably be common.